libstuff/SSignal.cpp

```diff
@@ -246,16 +246,22 @@ void _SSignal_StackTrace(int signum, siginfo_t *info, void *ucontext) {
                 }
                 free(frame);
             }
-            close(fd);
-
             // Done.
             free(callstack);
 
             // Call our die function and then reset it.
             SWARN("Calling DIE function.");
-            SSignalHandlerDieFunc();
-            SSignalHandlerDieFunc = [](){};
+            string logMessage = SSignalHandlerDieFunc();
+            SSignalHandlerDieFunc = [](){ return ""; };
             SWARN("DIE function returned.");
+            
+            // Finish writing the crash file with the request details if it exists
+            if (fd != -1 && !logMessage.empty()) {
+                logMessage = format("{}{}", logMessage, "\n");
```

Comment:
IMO this usage of `format` is confusing. Why not just do `logMessage += "\n"`?

Reply:
Changed.